tests/components/husqvarna_automower/test_init.py

```diff
@@ -127,21 +128,74 @@ async def test_websocket_not_available(
     caplog: pytest.LogCaptureFixture,
     freezer: FrozenDateTimeFactory,
 ) -> None:
-    """Test trying reload the websocket."""
-    mock_automower_client.start_listening.side_effect = HusqvarnaWSServerHandshakeError(
-        "Boom"
-    )
-    await setup_integration(hass, mock_config_entry)
-    assert "Failed to connect to websocket. Trying to reconnect: Boom" in caplog.text
-    assert mock_automower_client.auth.websocket_connect.call_count == 1
-    assert mock_automower_client.start_listening.call_count == 1
-    assert mock_config_entry.state is ConfigEntryState.LOADED
-    freezer.tick(timedelta(seconds=2))
-    async_fire_time_changed(hass)
-    await hass.async_block_till_done()
-    assert mock_automower_client.auth.websocket_connect.call_count == 2
-    assert mock_automower_client.start_listening.call_count == 2
-    assert mock_config_entry.state is ConfigEntryState.LOADED
+    """Test trying to reload the websocket."""
+    # Patch DEFAULT_RECONNECT_TIME to 0 for the test
+    with patch(
+        "homeassistant.components.husqvarna_automower.coordinator.DEFAULT_RECONNECT_TIME",
+        new=0,
+    ):
+        # Simulate a WebSocket handshake error
+        mock_automower_client.auth.websocket_connect.side_effect = (
+            HusqvarnaWSServerHandshakeError("Boom")
+        )
+
+        # Setup integration and verify initial log message
+        await setup_integration(hass, mock_config_entry)
+        assert (
+            "Failed to connect to websocket. Trying to reconnect: Boom" in caplog.text
+        )
+
+        # Initial call count and range for reconnection attempts
+        mock_automower_client.reset_mock()
+        default_recursion_limit = sys.getrecursionlimit()
+        temporary_recursion_limit = 200
```

Comment:
I was thinking in the order of magnitude of 10.

Or we just skip testing against the recursion limit. It's not like we always test against the recursion limit when we have something that repeats.

Reply:
On my machine we already have 44 recursions while setting up the integration. I think it's better to skip the tests, although it was a lot of work.